test/middleware.test.js

```diff
@@ -178,82 +181,87 @@ test('middlewares with prefix', t => {
   instance.get('/prefix/inner', handler)
 
   instance.listen({ port: 0 }, err => {
-    t.error(err)
-    t.teardown(instance.server.close.bind(instance.server))
+    t.assert.ifError(err)
+    t.after(() => instance.server.close())
 
-    t.test('/', t => {
+    t.test('/', (t, d) => {
```

Comment:
the `d` thing is very confusing and we could await the test to avoid the parent test's `done`

```suggestion
    await t.test('/', (t, done) => {
```